vortex-array/Cargo.toml

```diff
@@ -49,6 +49,7 @@ rstest_reuse = { workspace = true, optional = true }
 rustc-hash = { workspace = true }
 serde = { workspace = true, features = ["derive"] }
 static_assertions = { workspace = true }
+vortex-arcref = { workspace = true }
```

Comment:
Might be worth just shipping this as a separate PR anyway.

Could post the answer back to here: https://users.rust-lang.org/t/arc-or-static-reference/99988